rp2040-hal/src/rtc.rs

```diff
@@ -1,3 +1,620 @@
 //! Real Time Clock (RTC)
 // See [Chapter 4 Section 8](https://datasheets.raspberrypi.org/rp2040/rp2040_datasheet.pdf) for more details
-// TODO
+
+pub use crate::clocks::RtcClock;
+
+use crate::clocks::Clock;
+use embedded_time::fixed_point::FixedPoint;
+use rp2040_pac::{RESETS, RTC};
+
+/// Structure containing date and time information
+pub struct DateTime {
```

Comment:
It might be worth using [chrono](https://docs.rs/chrono/0.4.19/chrono/) instead of re-implementing calendar functionality here. We could put it behind a feature gate to avoid the dependency if required - but ensure chrono is added with `no-default-features` to turn off the `std` feature.